plugins/UsersManager/tests/UI/UsersManager_AnonymousUser_spec.js

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * UsersManager screenshot tests.
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+describe('UsersManager_AnonymousUser', function () {
+  this.fixture = "Piwik\\Plugins\\UsersManager\\tests\\Fixtures\\AnonymousUser";
+
+  const url = "?module=UsersManager&action=index";
+
+  function getUserAccess(userName) {
```

Comment:
Does this silently fail if the selector can't be found or returns an empty string? Should we have some fallback value?

Reply:
As jQuery is used (to support the `contains()` selector part), this would happen if the selector fails:

```
  1) UsersManager_AnonymousUser
       should start with a known list of users:

           expected '' to equal 'No access'
            Url to reproduce: http://localhost/tests/PHPUnit/proxy/?module=UsersManager&action=index
            Screenshot of failure: /srv/matomo/tests/UI/processed-ui-screenshots/should_start_with_a_known_list_of_users_failure.png
            No captured console logs.
  
      + expected - actual

      +No access
      
  AssertionError: expected '' to equal 'No access'
      at Context.<anonymous> (/srv/matomo/plugins/UsersManager/tests/UI/UsersManager_AnonymousUser_spec.js:58:49)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

It breaks at the expectation level and generates the usual failure screenshot.

I'd argue we should get rid of jQuery spcial selector usage and clean up the HTML to allow simple queries (yeah, there would be xpath, but I don't want to be the first to mess with that). Something like `#usersManagerData [data-row-key="someUserName"]` to get the row. And remove all the duplicate id attributes at the same time. But that might be an endeavour to be properly scoped beforehand, to not eat up too much time.

Reply:
Ok, thanks for the confirmation it produces an empty string when selector not found, I think that is good enough for now. We can think about the selectors if we do some more UI refactoring in the future, it's not a bad idea.